modules/core/src/lifecycle/prop-types.js

```diff
@@ -1,29 +1,78 @@
-import assert from '../utils/assert';
-
 const TYPE_DEFINITIONS = {
   boolean: {
     validate(value, propType) {
       return true;
+    },
+    equal(value1, value2, propType) {
+      return Boolean(value1) === Boolean(value2);
     }
   },
   number: {
-    validateType(value, propType) {
-      return (
-        'value' in propType &&
-        (!('max' in propType) || Number.isFinite(propType.max)) &&
-        (!('min' in propType) || Number.isFinite(propType.min))
-      );
-    },
     validate(value, propType) {
       return (
         Number.isFinite(value) &&
         (!('max' in propType) || value <= propType.max) &&
         (!('min' in propType) || value >= propType.min)
       );
     }
+  },
+  color: {
+    validate(value, propType) {
```

Comment:
Is `compare` field assumed to be true for this case, also for other cases where it is not used? I also didn't see any doc changes, please add these details to docs in this or next PR.